File: reddio-csharp/reddio.api/V1/Rest/OrderListMessage.cs
using Newtonsoft.Json;

namespace Reddio.Api.V1.Rest
{
    public class OrderListMessage
    {
        [JsonProperty("stark_key")] public string StarkKey;
        [JsonProperty("contract_address")] public string ContractAddress;
        [JsonProperty("token_ids")] public string[]? TokenIds;

        public OrderListMessage()
        {
        }

        public OrderListMessage(string starkKey, string contractAddress, string[]? tokenIds)
        {
            StarkKey = starkKey;
            ContractAddress = contractAddress;
            TokenIds = tokenIds;
        }
    }
}

File: reddio-java/reddio-api/src/main/java/com/reddio/api/v1/rest/OrderListMessage.java
package com.reddio.api.v1.rest;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor(staticName = "of")
public class OrderListMessage {
    @JsonProperty("stark_key")
    private String starkKey;
    @JsonProperty("contract_address")
    private String contractAddress;
    @JsonProperty("limit")
    private Long limit;
    @JsonProperty("page")
    private Long page;
    @JsonProperty("direction")
    private Integer direction;
    @JsonProperty("token_ids")
    private List<Long> tokenIds;
    @JsonProperty("order_state")
    private OrderState orderState;
}
